apps/dhparam.c

```diff
@@ -242,10 +242,19 @@ int dhparam_main(int argc, char **argv)
         } else
 # endif
         {
-            if (informat == FORMAT_ASN1)
+            if (informat == FORMAT_ASN1) {
+                /*
+                 * We have no PEM header to determine what type of DH params it
```

Comment:
I think it is better for a few reasons:
- I see this as a bug fix that goes all the way back to 1.0.2. Adding a new option (i.e. feature) for backport for this seems too much to me.
- It would be quite a jarring user experience. It is well established behaviour (since 1.0.2) that you do not need to specify the dh params type. It would be really unexpected behaviour that you suddenly have to if your inform is DER. I think that would catch a lot of people out.
- Adding a new option assumes you know what type you have, which I suspect most users would not.